misc/python/materialize/cli/mzconduct.py

```diff
@@ -946,27 +947,115 @@ def run(self, comp: Composition, workflow: Workflow) -> None:
             raise Failed(f"Unable to kill container {container_id}: {e}")
 
 
-@Steps.register("chaos-delay-docker")
-class ChaosDelayDockerStep(WorkflowStep):
-    """Delay the incoming and outgoing network traffic for a Docker service.
+class ChaosNetemStep(WorkflowStep):
+    """Base class for running netem chaos against a Docker container.
 
-    Params:
-        service: Docker service to delay, will be used to grep for container id
-                 NOTE: service name must be unique to correctly match the container id
-        delay: milliseconds to delay network traffic (default: 100ms)
+    We use pumba (a chaos testing tool for Docker) to run the various netem tests.
+    pumba only supports Docker container names (not container ids), meaning that
+    we have to pass the exact container name to these steps. We should fix this in
+    the future.
     """
 
-    def __init__(self, service: str, delay: int = 100) -> None:
-        self._service = service
-        self._delay = delay
+    def __init__(self, duration: int):
+        self._duration = duration
 
     def run(self, comp: Composition, workflow: Workflow) -> None:
+        if self._duration == -1:
+            # If duration isn't provided, run for 7 days in a non-blocking thread.
+            duration = 10080
+        else:
+            duration = self._duration
+
+        cmd = self.get_cmd(duration).split()
+        netem_thread = Thread(target=self.threaded_netem, args=(cmd,), daemon=True)
+        netem_thread.start()
```

Comment:
I think it would be fine to leave this running in the foreground and just have the process not exit until it has finished, unless it's necessary to run multiple of these at once, in which case still threading but waiting for it at the end would allow checking the exit code.

Reply:
Adding a `netem_thread.join()` to the end of this function makes it block, preventing the workflow from moving forward. I will think about other daemon alternatives

Reply:
We could stick the thread on the composition and join at the very end, I think?

I'm not sure if there's a better way of doing it.